lib/controller.js

```diff
@@ -677,6 +677,31 @@ controller.connectToNetwork = function(opts) {
       if (security && securityOptions.indexOf(security) < 0) {
         return reject(`"${security}" is not a valid security option. Please choose on of the following: ${securityOptions.join(', ')}`);
       }
+      
+      if (security === 'wep') {
+        // WEP passphrases can be 10, 26, or 58 hexadecimal digits long.
+        // Match for hexadecimal characters:
+        if (isNaN(`0x${password}`)) {
+          return reject('Invalid passphrase: WEP keys must consist of hexadecimal digits, i.e. 0 through 9 and "a" through "f".');
+        }
+        // Then test for length:
+        const length = password.length;
+        if (length !== 10 || length !== 26 || length !== 58) {
+          return reject('Invalid passphrase: WEP keys must be 10, 26, or 58 digits long for 64-, 128-, and 256-bit WEP.');
+        }
+      }
+      
+      if (security === 'psk' || security === 'psk2') {
+        // WPA/WPA2-PSK passphrases can be 8-63 ASCII characters, or 64 hexadecimal digits.
+        // Match ASCII codes for all 127 printable ASCII characters:
+        const asciiRegex = /^[\x00-\x7F]{8,63}$/;
+        // Match exactly 64 hexadecimal digits:
+        const hexTest = !isNaN(`0x${password}`) && password.length === 64;
+        // Reject if both tests fail.
+        if (!asciiRegex.test(password) && !hexTest) {
+          return reject('Invalid passphrase: WPA/WPA2-PSK passkeys must be 8-63 ASCII characters, or 64 hexadecimal digits.');
+        }
```

Comment:
Specifically, I'd like to see links to the published protocols, in which things like ` if (length !== 10 || length !== 26 || length !== 58) {` are clearly defined. Thanks!!